test/test_timeout.rb

```diff
@@ -127,11 +153,11 @@ def test_handle_interrupt
     bug11344 = '[ruby-dev:49179] [Bug #11344]'
     ok = false
     assert_raise(Timeout::Error) {
-      Thread.handle_interrupt(Timeout::Error => :never) {
+      Thread.handle_interrupt(Timeout::ExitException => :never) {
         Timeout.timeout(0.01) {
           sleep 0.2
           ok = true
-          Thread.handle_interrupt(Timeout::Error => :on_blocking) {
+          Thread.handle_interrupt(Timeout::ExitException => :on_blocking) {
```

Comment:
This seems to add a new public interface `Timeout::ExitException`.
Does it need to be public?

Reply:
It does if we want to allow users to use `Thread.handle_interrupt` with it.  From @eregon's research, there are only 2 gems actually doing that, and whether it is actually needed is not known.  If we decide we don't want to support that, we could make `ExitException` a private constant.